### ISSUE TITLE
Specify remote connection settings programatically (MsBuild)

### ISSUE THREAD

**Issue by Sam13** [Microsoft/VSLinux#364]
For build automation I'd like to provide remote connection settings programatically.
Preferred way would be MsBuild parameters.

Is this already possible? I did not find any information about that

I'm currently trying to achieve that functionality by copying a predefined _store.xml_ to _%APPDATA%..\Local\Microsoft\Linux\User Data\3.0\store.xml_.
This does not work, I always get the error  _Could not connect to the remote system, user name or password is incorrect._.
I assume the encrypted passwords cannot be transferred to another user account?

**Comment by lukka** [comment]
@Sam13 thanks for the request. If it is possible my suggestion is to use Visual Studio to setup the credentials, hopefully you need to do this only once. 
It is not possible to setup those programmatically, my advice would be to submit a request for a command line utility for doing so (on [DevComm](https://developercommunity.visualstudio.com)).
Your assumption is right, you cannot copy the file to another machine and use that with another Windows account.

**Comment by Sam13** [comment]
@lukka that's very sad. 
I'm running Visual Studio inside Docker, so I'm not aware of how to run the GUI and set those options there.

This small missing feature makes the whole cross compilation process unusable to run in a docker based CI environment.

If I would write such a command line utility myself can you give me a starting point on which methods / classes / assemblies provide the functionality to alter the credentials file?

**Comment by itodirel** [comment]
we support specifying the remote machine in the command line to MSBUILD if you already added the credentials to Connection Manager, with /p:RemoteTarget=, however we do not allow adding credentials to Connection Manager programatically, there is a way to do it but it isn't documented, if you're blocked I'd be happy yo tell you offline, please mail us at vcpplinux-support@microsoft.com

**Comment by Sam13** [comment]
@itodirel Many thanks for your kind offer.
In meantime I figured out something myself. I'll get back to you if I need more support...

**Comment by itodirel** [comment]
thanks, closing now, please reopen or comment if you still need help

**Comment by esweet431** [comment]
We now have full support for programmatically adding and removing remote connections: https://docs.microsoft.com/en-us/cpp/linux/connectionmanager-reference?view=vs-2019

**Comment by itodirel** [comment]
We retired this issue list, here on Github. We now adopted and using the Developer Community (https://developercommunity.visualstudio.com/spaces/62/index.html), for Cross Plat and Linux experiences (and C++ all up generally), which we are actively monitoring. Please use the Developer Community, to report issues or suggestions for bugs or new experiences. Further discussions on this thread here on Github might be unmonitored. If there is further comments or context in this current issue, please create a new issue on the Developer Community and mention it there. For existing issues which we can reproduce, or anything that is actionable to us, we will provide an update right here, in this thread. For new suggestions or issues, which at this time are in our roadmap, or we believe and have evidence that it will broadly and/or significantly benefit everyone, we will open a suggestion ticket, with the issue or suggestion, and link it here.
